frontend/packages/wizard/src/confirm-action.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import {
+  Button,
+  ConfirmActionProps,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  TextField,
+  Typography,
+} from "@clutch-sh/core";
+
+interface ConfirmActionDialogProps extends ConfirmActionProps {
+  open: boolean;
+  onCancel: () => void;
+}
+
+const ConfirmAction: React.FC<ConfirmActionDialogProps> = ({
+  open,
+  title,
+  description,
+  confirmationText,
+  actionLabel,
+  onConfirm,
+  onCancel,
+}) => {
+  const [input, setInput] = React.useState("");
+
+  React.useEffect(() => setInput(""), [open]);
+
+  const handleConfirm = () => {
+    if (input === confirmationText) {
+      onConfirm();
+    }
+  };
+
+  return (
+    <Dialog title={title} open={open} onClose={onCancel}>
+      <DialogContent>
+        {typeof description === "string" ? (
+          <Typography variant="body1">{description}</Typography>
+        ) : (
+          description
+        )}
+        <TextField
+          label="Confirmation"
+          value={input}
+          onChange={e => setInput(e.target.value)}
+          fullWidth
+        />
```

Comment:
We may want to leave this as part of the description. We may prefer to have to some workflows with a confirmation but not necessarily requiring an input, so that should be left up to the implementer.